build/stage-build.yml

```diff
@@ -39,80 +42,95 @@
 jobs:
   - job: On_Windows_ReleaseNotes
     pool:
-      name: $(windowsPoolName)
+      vmImage : windows-2022
     variables:
       ArtifactName: ReleaseNotes_${{ parameters.applicationEnvironment }}
       ApplicationEnvironment: ${{ parameters.applicationEnvironment }}
-
-    container: windows
-
     steps:
       - template: steps-build-release-notes.yml
         parameters:
           additionalReleaseNotesFile: ${{ parameters.additionalReleaseNotesFile }}
           removeHyperlinksFromReleaseNotes: ${{ parameters.removeHyperlinksFromReleaseNotes }}
 
-  - job: On_Windows # Build UWP and Android on a windows machine
+  - job: OnWindows_Windows
     strategy:
-      maxParallel: 3
       matrix:
-        Android:
-          ApplicationConfiguration: Release_Android
-          ApplicationPlatform: Any CPU
-          ArtifactName: $(AndroidArtifactName)_${{ parameters.applicationEnvironment }}
-          ApplicationEnvironment: ${{ parameters.applicationEnvironment }}
-        UWP:
-          ApplicationConfiguration: Release_UWP
-          ApplicationPlatform: Any CPU
-          ArtifactName: $(UWPArtifactName)_${{ parameters.applicationEnvironment }}
+        Windows:
+          ApplicationPlatform: x64
+          ArtifactName: $(WindowsArtifactName)_${{ parameters.applicationEnvironment }}
           ApplicationEnvironment: ${{ parameters.applicationEnvironment }}
+          BuildTargetFramework: net6.0-windows10.0.19041.0
+    pool:
+      vmImage : windows-2022
+    steps:
+      - template: steps-build-windows.yml
+        parameters:
+          pathToSrc: ${{ parameters.pathToSrc }}
+          solutionFileName: '${{ parameters.solutionName }}.sln'
+          solutionName: ${{ parameters.solutionName }}
+
+  - job: OnWindows_Tests
+    strategy:
+      matrix:
         Tests:
-          ApplicationConfiguration: Release_Tests
-          ApplicationPlatform: Any CPU
           ArtifactName: $(TestsArtifactName)_${{ parameters.applicationEnvironment }}
           ApplicationEnvironment: ${{ parameters.applicationEnvironment }}
+    pool:
+      vmImage : windows-2022
+    steps:
+      - template: steps-build-tests.yml
+        parameters:
+          pathToSrc: ${{ parameters.pathToSrc }}
+          solutionFileName: '${{ parameters.solutionName }}.sln'
+          solutionName: ${{ parameters.solutionName }}
+
+  - job: OnWindows_Android
```

Comment:
As this job can take more than an hour to complete (due to AOT, and probably we need to improve that), you can set a higher limit than the default 60 minutes through `timeoutInMinutes`.

```yml
- job: OnWindows_Android
  timeoutInMinutes: 90             
  strategy:
    matrix:
      Android:
        ApplicationEnvironment: ${{ parameters.applicationEnvironment }}
        ArtifactName: $(AndroidArtifactName)_${{ parameters.applicationEnvironment }}
        ApplicationPlatform: AnyCPU
        BuildForPlayStore: False
        BuildTargetFramework: net6.0-android
        VersionCodeOffset: 1
```